### PR TITLE
pip.list_all_versions now can read index_url

### PR DESCRIPTION
**Pull request by xyoun**
### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/45914

### Previous Behavior
pip.installed with custom index_url cause failed

### New Behavior
pip.list_all_versions now can read index_url

### Tests written?

No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: doc/man/salt.7 [hunk 1]
```diff
@@ -198363,7 +198363,7 @@ installed pip is new enough.
 .UNINDENT
 .INDENT 0.0
 .TP
-.B salt.modules.pip.list_all_versions(pkg, bin_env=None, include_alpha=False, include_beta=False, include_rc=False, user=None, cwd=None)
+.B salt.modules.pip.list_all_versions(pkg, bin_env=None, include_alpha=False, include_beta=False, include_rc=False, user=None, cwd=None, index_url=None)
```
Comment (rallytime) [inline doc/man/salt.7]: It's totally fine that these changes are here, but I wanted to let you know that we automatically generate the manpages for each release, so you don't need to update these in the future. :)

File: salt/modules/pip.py [hunk 2]
```diff
@@ -1334,6 +1335,9 @@ def list_all_versions(pkg,
     cwd
         Current working directory to run pip from
 
+    index_url
+        Base URL of Python Package Index
+
```
Comment (rallytime) [inline salt/modules/pip.py]: Can you add a `.. versionadded:: Fluorine` tag here?

**Review by rallytime — CHANGES_REQUESTED** [review]
I have one small request, otherwise this LGTM. @terminalmage Do you have anything to add?

**Review by terminalmage — APPROVED** [review]
Looks pretty straightforward.

File: doc/man/salt.7 (hunk 1 above)
Comment (xyoun) [reply doc/man/salt.7]: 😸

**Review by rallytime — APPROVED** [review]

**Pull request merged**